api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java

```diff
@@ -69,29 +69,89 @@ public void sendBetaAccessRequestEmail(String userName) throws MessagingExceptio
   }
 
   @Override
-  public void sendWelcomeEmail(String contactEmail, String password, User user)
+  public void sendWelcomeEmail(final String contactEmail, final String password, final User user)
       throws MessagingException {
-    try {
-      InternetAddress email = new InternetAddress(contactEmail);
-      email.validate();
-    } catch (AddressException e) {
-      throw new MessagingException("Email: " + contactEmail + " is invalid.");
-    }
-    MandrillMessage msg = buildWelcomeMessage(contactEmail, password, user);
+    final String msgHtml =
+        buildHtml(WELCOME_RESOURCE, welcomeMessageSubstitutionMap(password, user));
+    final MandrillMessage msg = buildMessage(contactEmail, msgHtml, "Your new All of Us Account");
     sendWithRetries(msg, String.format("Welcome for %s", user.getName()));
   }
 
   @Override
-  public void sendBetaAccessCompleteEmail(String contactEmail, String username)
+  public void sendBetaAccessCompleteEmail(final String contactEmail, final String username)
       throws MessagingException {
+    final String msgHtml = buildHtml(BETA_ACCESS_RESOURCE, betaAccessSubstitutionMap(username));
+    final MandrillMessage msg =
+        buildMessage(contactEmail, msgHtml, "All of Us ID Verification Complete");
+    sendWithRetries(msg, String.format("BetaAccess Complete for %s", contactEmail));
+  }
+
+  private ImmutableMap<String, String> welcomeMessageSubstitutionMap(
+      final String password, final User user) {
+    final CloudStorageService cloudStorageService = cloudStorageServiceProvider.get();
+    return new ImmutableMap.Builder<String, String>()
+        .put("USERNAME", user.getPrimaryEmail())
+        .put("PASSWORD", password)
+        .put("URL", workbenchConfigProvider.get().admin.loginUrl)
+        .put("HEADER_IMG", cloudStorageService.getImageUrl("all_of_us_logo.png"))
+        .put("BULLET_1", cloudStorageService.getImageUrl("bullet_1.png"))
+        .put("BULLET_2", cloudStorageService.getImageUrl("bullet_2.png"))
+        .build();
+  }
+
+  private ImmutableMap<String, String> betaAccessSubstitutionMap(final String username) {
```

Comment:
nit: for private methods this is OK, but generally the guidance is to declare your function to return Map (for compatibility), even when using an ImmutableMap under the hood. Which seems scary.